client/app/lib/helpers/download.js

```diff
@@ -0,0 +1,55 @@
+const DELAY = 500; // Interval to check for job completion in ms.
+
+function showSpinner($button) {
+  $button.addClass('disabled');
+  $button.children('.fa').hide();
+  const $spinner = $('<i class="fa fa-spinner fa-spin"></i>');
+  if ($button.children('.fa:not(.fa-lg)').length === 0) {
+    $spinner.addClass('fa-lg');
+  }
+  $button.append($spinner);
+}
+
+function hideSpinner($button) {
+  $button.removeClass('disabled');
+  $button.children('.fa').show();
+  $button.children('i.fa-spinner').remove();
+}
+
+function waitForJob(url, $button) {
+  setTimeout(() => {
+    $.ajax({
+      url,
+      method: 'GET',
+      dataType: 'json',
+      global: false,
+    }).done((response) => {
+      if (response.status === 'completed') {
+        if (response.redirect_url) {
+          window.location.href = response.redirect_url;
+        }
+        hideSpinner($button);
+      } else if (response.status === 'submitted') {
+        waitForJob(url, $button);
+      }
+    });
+  }, DELAY);
+}
+
+function onDownload(e) {
```

Comment:
How about name this `initializeDownloadLink($link)` ? And pass in a $link, and I think `$link` is less confusing that `$button` cause you need the `href`.